aspnet/security/authentication/cookieauthentication.rst

```diff
@@ -0,0 +1,149 @@
+
+.. code-block:: c#
+
+ public static class LastChangedValidator
+ {
+     public static async Task ValidateAsync(CookieValidatePrincipalContext context)
+     {
+         // Pull database from registered DI services.
+         var userRepository = context.HttpContext.RequestServices.GetRequiredService<IUserRepository>();
+         var userPrincipal = context.Principal;
+
+         // Look for the last changed claim.
+         string lastChanged;
+         lastChanged = (from c in userPrincipal.Claims
+                        where c.Type == "LastUpdated"
+                        select c.Value).FirstOrDefault();
+
+         if (string.IsNullOrEmpty(lastChanged) ||
+             !userRepository.ValidateLastChanged(userPrincipal, lastChanged))
+         {
+             context.RejectPrincipal();
+             await context.HttpContext.Authentication.SignOutAsync("MyCookieMiddlewareInstance");
+         }
+     }
+ }
+
+This would then be wired up during cookie middleware configuration
+
+.. code-block:: c#
+
+ app.UseCookieAuthentication(options =>
+ {
+     options.Events = new CookieAuthenticationEvents
+     {
+         // Set other options
+         OnValidatePrincipal = LastChangedValidator.ValidateAsync
+     };
+ });
+
+If you want to non-destructively update the user principal, for example, their name might have been updated, a decision which doesn't affect security in any way you can call ``context.ReplacePrincipal()`` and set the ``context.ShouldRenew`` flag to ``true``.
+
+.. _security-authentication-cookie-options:
+
+Controlling cookie options
+--------------------------
+
+The ``CookieAuthenticationOptions`` class comes with various configuration options to enable you to fine tune the cookies created.
+
+* CookieDomain - the domain name the cookie will be served to. By default this is the host name the request was sent to. The browser will only serve the cookie to a matching host name. You may wish to adjust this to have cookies available to any host in your domain. For example setting the cookie domain to ``.contoso.com`` will make it available to ``contoso.com``, ``www.contoso.com``, ``staging.www.contoso.com`` etc.
+* CookieHttpOnly - a flag indicating if the cookie should only be accessible to servers. This defaults to ``true``. Changing this value may open your application to cookie theft should your application have a Cross Site Scripting bug.
+* CookieSecure - a flag indicating if the cookie created should be limited to HTTPS, HTTP or HTTPS, or the same protocol as the request. This defaults to ``SameAsRequest``. 
+* ExpireTimeSpan - the ``TimeSpan`` after which the cookie will expire. This is added to the current date and time to create the expiry date for the cookie.
+* SlidingExpiration - a flag indicating if the cookie expiration date will be reset after each request. The new expiry date will be moved forward to be the current date plus the ``ExpireTimespan``.
```

Comment:
Actually, `will be reset after each request` is not true. When sliding expiration is enabled, a new cookie is issued only when the remaining time is less than the elapsed time.